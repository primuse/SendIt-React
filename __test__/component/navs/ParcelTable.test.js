import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import {
  ParcelTable, DeliveredParcelTable, InTransitParcelTable
} from '../../../src/components/navs/table.jsx';

const parcels = [
  {
    id: '1',
    parcelName: 'rice',
    weigth: '30',
    price: '4,000',
    destination: 'Owerri',
    receiver: 'Tiku Okoye',
    senton: '22-03-2019',
    status: 'created'
  },
  {
    id: '1',
    parcelName: 'rice',
    weigth: '30',
    price: '4,000',
    destination: 'Owerri',
    receiver: 'Tiku Okoye',
    senton: '22-03-2019',
    status: 'created'
  }
];


describe('Parcel Table component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <ParcelTable parcels={parcels} />
      </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Delivered Parcel Table component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <DeliveredParcelTable parcels={parcels} />
      </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('In-Transit Parcel Table component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <InTransitParcelTable parcels={parcels} />
      </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
