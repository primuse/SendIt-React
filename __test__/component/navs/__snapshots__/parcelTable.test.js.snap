// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parcel Table component renders correctly 1`] = `
<div
  className="dash-cont"
>
  <div
    id="table-info"
  >
    <a
      href="#"
    />
  </div>
  <div
    id="table-cont"
  >
    <table>
      <tbody>
        <tr
          id="table-head"
        >
          <th>
            ID
          </th>
          <th>
            Name
          </th>
          <th>
            Weight
          </th>
          <th>
            Price
          </th>
          <th>
            Destination
          </th>
          <th>
            Receiver
          </th>
          <th>
            Sent On
          </th>
          <th>
            Status
          </th>
        </tr>
        <tr>
          <td>
            PO
            1
          </td>
          <td />
          <td>
            kg
          </td>
          <td>
            N
            4,000
          </td>
          <td>
            Owerri
          </td>
          <td>
            Tiku Okoye
          </td>
          <td>
            22-03-2019
          </td>
          <td>
            created
          </td>
          <td>
            <a
              className="btn xsm bg-bright-blue white"
              href="/details/1"
              onClick={[Function]}
            >
              View
            </a>
          </td>
        </tr>
        <tr>
          <td>
            PO
            1
          </td>
          <td />
          <td>
            kg
          </td>
          <td>
            N
            4,000
          </td>
          <td>
            Owerri
          </td>
          <td>
            Tiku Okoye
          </td>
          <td>
            22-03-2019
          </td>
          <td>
            created
          </td>
          <td>
            <a
              className="btn xsm bg-bright-blue white"
              href="/details/1"
              onClick={[Function]}
            >
              View
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
